### PR TITLE
fix font rendering position

### PR DESCRIPTION
**Pull request by newnon**
fix font always padded on 2 pixels left

### CODE REVIEW

File: cocos/2d/CCFontAtlas.cpp [hunk 1]
```diff
@@ -353,7 +353,7 @@ bool FontAtlas::prepareLetterDefinitions(const std::u16string& utf16Text)
             tempDef.validDefinition = true;
             tempDef.width = tempRect.size.width + _letterPadding + _letterEdgeExtend;
             tempDef.height = tempRect.size.height + _letterPadding + _letterEdgeExtend;
-            tempDef.offsetX = tempRect.origin.x + adjustForDistanceMap + adjustForExtend;
+            tempDef.offsetX = tempRect.origin.x + adjustForDistanceMap - adjustForExtend;
```
Comment (dumganhar) [inline cocos/2d/CCFontAtlas.cpp]: Should `adjustForDistanceMap` also be negative like:

``` c++
tempDef.offsetX = tempRect.origin.x - adjustForDistanceMap - adjustForExtend;
```

BTW, how did you find this issue? Any test case for it?

**Comment by dumganhar** [comment]
This patch fixed the following issues:
https://github.com/cocos2d/cocos2d-x/issues/16399
https://github.com/cocos2d/cocos2d-x/issues/16663

Thanks @newnon .

**Comment by halx99** [comment]
@dumganhar  I applied this PR, and the test result is looks good:
wf342o3 w2m4dnqbcqhhqp

**Comment by dumganhar** [comment]
@halx99 , got it, thanks for testing.

File: cocos/2d/CCFontAtlas.cpp (hunk 1 above)
Comment (dumganhar) [reply cocos/2d/CCFontAtlas.cpp]: @newnon, could you reply to my question? Thanks.
Should adjustForDistanceMap also be negative?

Comment (newnon) [reply cocos/2d/CCFontAtlas.cpp]: Looks like it should, i changed this PR

**Comment by dumganhar** [comment]
@newnon , I also added a test case for this patch in this PR https://github.com/cocos2d/cocos2d-x/pull/16703.

Your Pull Request will be automatically merged while mine is merged. Thanks.

**Comment by minggo** [comment]
close this PR since #16703 is merged

**Pull request merged**